java/yb-cql/src/test/java/org/yb/cql/CQLTester.java

```diff
@@ -0,0 +1,187 @@
+    {
+        String currentTable = createTableName();
+        String fullQuery = formatQuery(keyspace, query);
+        final long startTimeMs = System.currentTimeMillis();
+        session.execute(fullQuery);
+        long elapsedTimeMs = System.currentTimeMillis() - startTimeMs;
+        LOG.info("After execute: {} took {}ms", fullQuery, elapsedTimeMs);
+    }
+
+    protected String createTableName()
+    {
+        String currentTable = "table_" + seqNumber.getAndIncrement();
+        tables.add(currentTable);
+        return currentTable;
+    }
+
+    protected String formatQuery(String query)
+    {
+        return formatQuery(KEYSPACE, query);
+    }
+
+    protected final String formatQuery(String keyspace, String query)
+    {
+        String currentTable = currentTable();
+        return currentTable == null ? query : String.format(query, keyspace + "." + currentTable);
+    }
+
+    protected String keyspace()
+    {
+        return KEYSPACE;
+    }
+
+    protected String currentTable()
+    {
+        if (tables.isEmpty())
+            return null;
+        return tables.get(tables.size() - 1);
+    }
+
+    public ResultSet execute(String query, Object... values) throws Exception {
+        String fullQuery = formatQuery(KEYSPACE, query);
+        final ResultSet result = session.execute(fullQuery, values);
+        LOG.info("EXEC CQL FINISHED: {}", fullQuery);
+        return result;
+    }
+
+
+    public ResultSet execute(String query) throws Exception {
+        String fullQuery = formatQuery(KEYSPACE, query);
+        final ResultSet result = session.execute(fullQuery);
+        LOG.info("EXEC CQL FINISHED: {}", fullQuery);
+        return result;
+    }
+
+    protected void assertAllRows(Object[]... rows) throws Throwable
+    {
+        assertRows(execute("SELECT * FROM %s"), rows);
+    }
+
+    public static Object[] row(Object... expected)
+    {
+        return expected;
+    }
+
+    public static int displayRows(ResultSet result)
+    {
+        if (result == null)
+        {
+            LOG.info("Resultset has zero rows");
+            return 0;
+        }
+
+        ColumnDefinitions colDef = result.getColumnDefinitions();
+        List<ColumnDefinitions.Definition> meta = colDef.asList();
+        int i = 0;
+        for (Row actual : result)
+        {
+            LOG.info("Row {} = {}", i, actual.toString());
+            i++;
+        }
+	return i;
+    }
+
+    public static void assertRows(ResultSet result, Object[]... rows)
+    {
+        if (result == null)
+        {
+            if (rows.length > 0)
+                fail(String.format("No rows returned by query but %d expected", rows.length));
+            return;
+        }
+        ColumnDefinitions colDef = result.getColumnDefinitions();
+        List<ColumnDefinitions.Definition> meta = colDef.asList();
+        Iterator<Row> iter = result.iterator();
+        int i = 0;
+        if (!iter.hasNext() && rows.length > 0)
```

Comment:
This check is unnecessary. The check `if (i != rows.length) ...` after the `while` loop below does the same check essentially.

Reply:
Thanks @robertpang. This check was raising "No rows returned by query..." error, to not loose this error case, I changed the failure message for the `if (i != rows.length)` to `fail(String.format("Query returned %d rows but expected %d rows", i, rows.length));`

Thanks!